### PR TITLE
Semiautomatic update `.env` from `env.template`

### PR DESCRIPTION
**Pull request by Artemonim**
### Background
A user can make a `git pull`, not check for changes and not find out about changes in settings.

### Changes
I've add `update_env_from_template(self)` to the `config.py`

### Documentation
This is a Python function that updates a configuration file `.env` from a template file `.env.template`.
The function performs the following steps:
1. Finds paths to the `.env` and `.env.template` files.
2. Checks if the `.env.template` file exists. If not, prints an error and exits the function.
3. Checks if the `.env` file exists. If not, prompts the user to create an empty file or prints an error and exits the function.
4. Loads the values from both files.
5. Compares the values and identifies the keys that were removed from the `.env.template` file and the values that need to be added to the `.env` file from the template.
6. Prompts the user to confirm the changes. If confirmed, updates the `.env` file with new or changed values and removes the keys that were removed from the `.env.template` file.
7. Prints any errors that occur during this process.

### Test Plan
GPTmachines, son 😎

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thouroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Comment by Artemonim** [comment]
TODO: Create .env for the first launch

**Review by nponeccop — APPROVED** [review]

**Review by nponeccop — DISMISSED** [review]

**Comment by nponeccop** [comment]
@Artemonim Now there are conflicts

**Comment by Artemonim** [comment]
@nponeccop Test failed, but I can't understand why

**Comment by nponeccop** [comment]
What failed was not the tests. It was flake8, a Python linter. 

> scripts/config.py:92:5: E303 too many blank lines (2)

You are supposed to delete one line in config.py, lamely the line 92, to make 1 consecutive blank line instead of 2.

Yes, this sort of contradicts the earlier requirement of not making whitespace changes. Now you are supposed to make changes to satisfy flake8. You can get its command-line by looking at https://github.com/Torantulino/Auto-GPT/blob/master/.github/workflows/ci.yml#L35

It's flake8 scripts/ tests/ --select E303,W293,W291,W292,E305

But IMO a saner way to do it is to just run

python -m flake8 --select E303,W293,W291,W292,E305 

in the repo root.

You satisfy that command, then push.

**Comment by nponeccop** [comment]
Now the tests fail, too. @p-i-

**Comment by Artemonim** [comment]
Unfortunately, I don’t have time for that right now. I wrote a workaround with ChatGPT, but my function still gets called somewhere.

**Review by nponeccop — DISMISSED** [review]

**Comment by Artemonim** [comment]
May be totally broken after #980
I'm scared to test it 💀

File: scripts/main.py [hunk 1]
```diff
@@ -18,7 +18,6 @@
 import logging
 from prompt import get_prompt
 
-cfg = Config()
```
Comment (nponeccop) [inline scripts/main.py]: This is wrong (probably from a merge). Isn't cfg used below on line 26?

**Review by nponeccop — CHANGES_REQUESTED** [review]

**Comment by nponeccop** [comment]
And the linter is red

**Comment by Artemonim** [comment]
I fixed it to the state of not passing the test. Yay!

**Comment by Artemonim** [comment]
I have never worked with unit tests in Python. If someone can tell me what is now calling the function to bypass the flag, I would appreciate it.

**Review by nponeccop — APPROVED** [review]

**Comment by nponeccop** [comment]
@Artemonim Now linting failed. We gradually tighten the linting once a day

**Comment by Artemonim** [comment]
> @Artemonim Now linting failed. We gradually tighten the linting once a day

I can only see the unit test failure

**Comment by nponeccop** [comment]
> > @Artemonim Now linting failed. We gradually tighten the linting once a day
> 
> I can only see the unit test failure

My bad

**Comment by nponeccop** [comment]
@Artemonim CI fails now

**Comment by Artemonim** [comment]
I tried 🥲

**Comment by nponeccop** [comment]
@Artemonim There are conflicts now

**Comment by Artemonim** [comment]
> @Artemonim There are conflicts now

We can't merge my PR anyway without someone to help me make or bypass the unit tests.
- The problem in bypassing is that the import calls `Config()` which does not contain the test bypass key.
- I am not qualified enough to write the tests myself. The latest commits are only the most I could do with the help of ChatGPT.

**Comment by nponeccop** [comment]
@richbeales @BillSchumacher Can you help @Artemonim ?

**Comment by Artemonim** [comment]
With `pytest --cov=autogpt --without-integration --without-slow-integration` I have 41 test passed, 7 skipped and 2 warnings on my local machine

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

**Comment by Pwuts** [comment]
Looks useful; needs rebasing

File: scripts/main.py (hunk 1 above)
Comment (Wladastic) [reply scripts/main.py]: Correct, should be added back in.

**Comment by Swiftyos** [comment]
Closing old PRS

**Pull request closed without merging**